src/renderers/dom/fiber/ReactDOMFiber.js

```diff
@@ -197,6 +212,18 @@ var DOMRenderer = ReactFiberReconciler({
     return true;
   },
 
+  commitInitialEffects(
+    domElement : Instance,
+    type : string,
+    newProps : Props,
+    rootContainerInstance : Container,
+    internalInstanceHandle : Object,
+  ) : void {
+    if (shouldAutoFocusHostComponent(type, newProps)) {
+      (domElement : any).focus();
+    }
```

Comment:
> In fact the code to handle this is already in `master`. You deleted it in a previous commit but I think if you add it back, it will work now that the host component is scheduled to update on initial mount.

The previous code was in `setInitialProperties` which is only called by `finalizeInitialChildren` which happens too early (during _complete_ not _commit_).

If you are referring to moving it into `updateDOMProperties`, then correct me if I'm wrong, but doing that would cause us to call `domElement.focus()` more than just on initial mount- which is not the way `autoFocus` should work.